### PR TITLE
[BSP] Support YD-CH32V307VCT6

### PR DESCRIPTION
**Pull request by GSunwinder**
## 拉取/合并请求描述：(PR description)

[

#### 请提供验证的bsp和config (provide the config and bsp) 

- BSP: bsp/wch/risc-v/yd-ch32v307vct6

- .config:

- action:

]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 代码是高质量的 Code in this PR is of high quality
- [x] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

File: bsp/wch/risc-v/yd-ch32v307vct6/applications/main.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2024, GSunwinder
```
Comment (supperthomas) [inline bsp/wch/risc-v/yd-ch32v307vct6/applications/main.c:2]: ？

**Review by supperthomas — CHANGES_REQUESTED** [review]

File: bsp/wch/risc-v/yd-ch32v307vct6/applications/main.c (hunk 1 above)
Comment (GSunwinder) [reply bsp/wch/risc-v/yd-ch32v307vct6/applications/main.c:2]: I don't understand the question, but if you want to change the license, you can install the correct one.

Comment (mysterywolf) [inline bsp/wch/risc-v/yd-ch32v307vct6/applications/main.c:2]: Hi please use this comment
```suggestion
 * Copyright (c) 2006-2022, RT-Thread Development Team
```

**Review by mysterywolf — CHANGES_REQUESTED** [review]

File: bsp/wch/risc-v/yd-ch32v307vct6/applications/main.c (hunk 1 above)
Comment (GSunwinder) [reply bsp/wch/risc-v/yd-ch32v307vct6/applications/main.c]: thanks for the tip

**Review by mysterywolf — APPROVED** [review]

File: bsp/wch/risc-v/yd-ch32v307vct6/rtconfig.h [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+/* Micrium: Micrium software products porting for RT-Thread */
+
+#define PKG_USING_CHERRYUSB
+#define PKG_CHERRYUSB_DEVICE
+#define PKG_CHERRYUSB_DEVICE_FS
+#define PKG_CHERRYUSB_DEVICE_CH32
+#define PKG_CHERRYUSB_DEVICE_CH32_CH32V307
+#define PKG_CHERRYUSB_DEVICE_CDC
+#define PKG_USING_CHERRYUSB_LATEST_VERSION
+
```
Comment (mysterywolf) [inline bsp/wch/risc-v/yd-ch32v307vct6/rtconfig.h:192]: Please don't select the software package by default.
Not simply remote those macros, please use menuconfig to cancel the software package and upload the rtconfig.h and .config which are re-generated.

File: bsp/wch/risc-v/yd-ch32v307vct6/rtconfig.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    POST_ACTION = OBJCPY + ' -O binary $TARGET ' + TARGET_NAME + '\n'
+    POST_ACTION += SIZE + ' $TARGET\n'
+
+def dist_handle(BSP_ROOT, dist_dir):
+    import sys
+    cwd_path = os.getcwd()
+    sys.path.append(os.path.join(os.path.dirname(BSP_ROOT), 'tools'))
+    from sdk_dist import dist_do_building
+    dist_do_building(BSP_ROOT, dist_dir)
```
Comment (mysterywolf) [inline bsp/wch/risc-v/yd-ch32v307vct6/rtconfig.py:70]: please press the enter button to create a new line.

**Review by mysterywolf — CHANGES_REQUESTED** [review]

**Review by mysterywolf — APPROVED** [review]

**Review by supperthomas — APPROVED** [review]

**Comment by mysterywolf** [comment]
Thanks for the PR!

**Pull request merged**